File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
         int el;
         int cnt=0;
         int n= nums.size();
        for(int i=0;i<n;i++){
           if(cnt<=0){
            el=nums[i];
            cnt=1;
           }
           else if(el==nums[i]){
                cnt++;
            }
            else{
                cnt--;
            }
        }
        if(cnt>0) {
            return el;
        }
        return -1;
        
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        dici={}
        for i in range(len(nums)):
            temp=1
            if nums[i] in dici:
                dici[nums[i]]=dici[nums[i]]+1
            else:
                dici[nums[i]]=temp
        print(dici)
       
        for i in dici:
            if dici[i] >len(nums)/2:

                print(i)
                return i
            # else:
            #     i=i+1